### PR TITLE
Add form for custom software install requests

### LINKED ISSUE #341: Web form for HPC Software Install
This should not be integrated into the default Hippo workflow .. yet, but we do need it to be CAS enabled.

**Cluster Name:** 
<selection field, with options>
-Farm
-Franklin
-HPC1
-HPC2
-LSSC0
-Peloton

**UC Davis or Affiliate Email:**
<prepopulated with CAS email, editable>

**Account Name on HPC Cluster**
<prepopulated with Kerberos ID, editable>

**Software Title and Version Requested:**
<text field>

**Software License**
<text field>

**Software Home Page:**
<text field>

**How the software will benefit others by making it available to all users:**
<text area>

**Any further information needed?**
<text area>

**Hidden Fields:**
CAS Login ID

**Email to user after submission:**

"Thank you for submitting your request. We appreciate your interest and will begin evaluating the details provided. Our team will review your submission thoroughly and respond as soon as possible.

If we need any further information, we will contact you directly.

Thank you for your patience."

**Email to SNOW after submission to hpc-help@ucdavis.edu** 
Include Web Fields, including the following in the body for consumption by SNOW:

Category: Request
Subcategory: New
Caller: [CAS Login ID]
ConfigurationItem: HPC Software

### PR DESCRIPTION
**Pull request by sprucely**
Currently there are no navigation links to the form, but it is accessible to authenticated users at `/softwarerequest`

Email to hpc-help@ucdavis.edu
image

Email to requester
image

Closes #341

### CODE REVIEW

**Comment by jSylvestre** [comment]
Depends what they want, but I'd tend to put the routing stuff for ServiceNow at the very top.

File: Hippo.Email/Models/SimpleNotificationModel.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Hippo.Email.Models
+{
+    public class SimpleNotificationModel
+    {
+        public string UcdLogoUrl { get; set; } = "https://hippo-test.azurewebsites.net/media/caes-logo-gray.png";
```
Comment (jSylvestre) [inline Hippo.Email/Models/SimpleNotificationModel.cs:11]: Don't have it point to test.

File: Hippo.Web/ClientApp/src/components/ClusterAdmin/SoftwareRequestForm.tsx [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+            className="form-control"
+            value={softwareRequestModel.email}
+            onChange={(e) =>
+              setSoftwareRequestModel({
+                ...softwareRequestModel,
+                email: e.target.value,
+              })
+            }
+          />
+          <p className="form-helper">Enter your email address.</p>
```
Comment (jSylvestre) [inline Hippo.Web/ClientApp/src/components/ClusterAdmin/SoftwareRequestForm.tsx:102]: Should we allow them to do this, or get this info from IAM?

File: Hippo.Web/Controllers/SoftwareController.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        // generate plain text email that will be forwarded to ServiceNow
+        var emailModel = new EmailModel
+        {
+            Emails = new string[] { "hpc-help@ucdavis.edu" },
+            Subject = "HPC Software Install Request",
+            TextBody = @$"Category: Request
+
+Subcategory: New
+
+Caller: {currentUser.Kerberos}
```
Comment (jSylvestre) [inline Hippo.Web/Controllers/SoftwareController.cs:45]: Name for IAM or an email

File: Hippo.Web/Controllers/SoftwareController.cs [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+    [HttpPost]
+    public async Task<ActionResult> RequestInstall([FromBody] SoftwareRequestModel softwareRequestModel)
+    {
+        if (!ModelState.IsValid)
+        {
+            return BadRequest(ModelState);
+        }
+
+        var currentUser = await _userService.GetCurrentUser();
```
Comment (jSylvestre) [inline Hippo.Web/Controllers/SoftwareController.cs:34]: Might not be good enough (at least if you use the name), I think if it is in the DB we don't pull from IAM? And only IAM will have the latest name.

**Review by jSylvestre — CHANGES_REQUESTED** [review]
Didn't test, just looked through the code.

File: Hippo.Email/Models/SimpleNotificationModel.cs (hunk 1 above)
Comment (sprucely) [reply Hippo.Email/Models/SimpleNotificationModel.cs:11]: Copied that from elsewhere. Going to have to make a couple changes...

File: Hippo.Web/ClientApp/src/components/ClusterAdmin/SoftwareRequestForm.tsx (hunk 2 above)
Comment (sprucely) [reply Hippo.Web/ClientApp/src/components/ClusterAdmin/SoftwareRequestForm.tsx:102]: It defaults to the value from IAM

File: Hippo.Web/Controllers/SoftwareController.cs (hunk 3 above)
Comment (sprucely) [reply Hippo.Web/Controllers/SoftwareController.cs:45]: Adding a Typeahead to search for user...

**Review by jSylvestre — APPROVED** [review]
Didn't test.
But as we are not showing a link, we can fix after testing if needed.

**Pull request merged**